### DISCUSSION TITLE
Expensive queries / caching

### DISCUSSION THREAD

**Discussion by Alexandre789010** [abpframework/abp#16139] [Q&A] [answered]
Hi,

I'm working with some large tables that I need to query and retrieve data from to generate reports. The data is updated only once a day, but I have to access it multiple times. I'm not very familiar with caching, and I'm hoping someone can provide some guidance on how to tackle this issue.

Would caching the list of data I obtain from the query be a feasible solution? Can I accomplish this using ABP and Redis?

Thanks for your time 😄

**Comment by maliming** [comment, marked as answer]
hi

You can start from document.

https://docs.abp.io/en/abp/latest/Caching
https://docs.abp.io/en/abp/latest/Redis-Cache
https://docs.abp.io/en/abp/latest/Entity-Cache
